### PR TITLE
Added DashboardLogic component

### PR DESCRIPTION
**Pull request by cheina97**
Components added:

- DashboardLogic

This PR add the DashboardLogic component, whose purpose is to get the informations about the users tenant workspaces and pass them to the Dashboard component.

### CODE REVIEW

File: frontend/src/components/workspaces/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -3,30 +3,42 @@ import { Col } from 'antd';
...
   return (
     <>
       <Col span={24} lg={8} xxl={8} className="lg:pr-4 py-5 lg:h-full flex">
         <div className="flex-auto lg:overflow-x-hidden overflow-auto scrollbar">
           <WorkspaceGrid
-            selectedWs={selectedWs}
-            workspaceItems={workspaceItems}
+            selectedWs={selectedWsId}
+            workspaceItems={workspaces.map((wk, idx) => ({
+              key: idx,
```
Comment (GabriFila) [inline frontend/src/components/workspaces/Dashboard/Dashboard.tsx:25]: Why use both a `key` and and `id`?

File: frontend/src/components/workspaces/Grid/WorkspaceGridItem/WorkspaceGridItem.tsx [hunk 2]
```diff
@@ -18,7 +18,9 @@ const WorkspaceGridItem: FC<IWorkspaceGridItemProps> = ({ ...props }) => {
           className={`cursor-pointer row shadow-lg h-24 w-24 2xl:h-28 2xl:w-28 workspaceitem ${
             isActive ? 'active' : ''
           }`}
-          onClick={() => onClick(id)}
+          onClick={() => {
```
Comment (GabriFila) [inline frontend/src/components/workspaces/Grid/WorkspaceGridItem/WorkspaceGridItem.tsx:21]: If it is active I would do nothing -> call the onClick only if it is not active

File: frontend/src/components/workspaces/Dashboard/Dashboard.tsx (hunk 1 above)
Comment (cannarelladev) [reply frontend/src/components/workspaces/Dashboard/Dashboard.tsx]: fixed

File: frontend/src/components/workspaces/Grid/WorkspaceGridItem/WorkspaceGridItem.tsx (hunk 2 above)
Comment (cannarelladev) [reply frontend/src/components/workspaces/Grid/WorkspaceGridItem/WorkspaceGridItem.tsx]: fixed

**Review by GabriFila — APPROVED** [review]

**Pull request merged**